server/channels/web/handlers.go

```diff
@@ -180,6 +181,10 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			responseLogFields = append(responseLogFields, mlog.String("status_code", statusCode))
 		}
 		mlog.Debug("Received HTTP request", responseLogFields...)
+
+		if !rateLimitExceeded {
+			h.recordMetrics(c, w, r, now)
+		}
```

Comment:
Actually, just a minor bug here. The `statusCode` variable needs to be set for `if statusCode != 0` check to work. And that used to work before because this got set in L386. 

Something like this could work:

```diff
--- a/server/channels/web/handlers.go
+++ b/server/channels/web/handlers.go
@@ -176,6 +176,7 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
                if c.AppContext.Session() != nil {
                        responseLogFields = append(responseLogFields, mlog.String("user_id", c.AppContext.Session().UserId))
                }
+               statusCode = strconv.Itoa(w.(*responseWriterWrapper).StatusCode())
                // Websockets are returning status code 0 to requests after closing the socket
                if statusCode != "0" {
                        responseLogFields = append(responseLogFields, mlog.String("status_code", statusCode))
@@ -183,7 +184,7 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
                mlog.Debug("Received HTTP request", responseLogFields...)

                if !rateLimitExceeded {
-                       h.recordMetrics(c, w, r, now)
+                       h.recordMetrics(c, r, now, statusCode)
                }
        }()

@@ -392,8 +393,7 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
        }
 }

-func (h Handler) recordMetrics(c *Context, w http.ResponseWriter, r *http.Request, now time.Time) {
-       statusCode := strconv.Itoa(w.(*responseWriterWrapper).StatusCode())
+func (h Handler) recordMetrics(c *Context, r *http.Request, now time.Time, statusCode string) {
        if c.App.Metrics() != nil {
                c.App.Metrics().IncrementHTTPRequest()
```

Reply:
Gotcha. Since the `statuscode` variable was being used only inside the defer block, I moved it completely in there.